runtime/vm/jnicsup.cpp

```diff
@@ -1874,7 +1874,7 @@ getMethodOrFieldID(JNIEnv *env, jclass classReference, const char *name, const c
 
 		/* ensure that the class is initialized */
 		if (declaringClass->initializeStatus != J9ClassInitSucceeded && declaringClass->initializeStatus != (UDATA) vmThread) {
-			gpCheckInitialize(vmThread, declaringClass);
+			gpCheckInitialize(vmThread, 0 == strcmp(name, "main") ? clazz : declaringClass);
```

Comment:
I feel like this is too arbirtary. I dont think "main" should dicate the behaviour.

I can see us needing to initialize both clazz and declaringClass (in that order). In most cases they are usually the same.

Perhaps we need more experimentation to verify this behaviour.

Reply:
Should this behaviour be determined by the call to get the `mainID` in the java launcher, or should it be the default behaviour? Additionally, what tests would be good to verify this behaviour?

Reply:
We need to test if RI will always initialize the declaring class or the named class (`clazz` inthis case). You can try examples that have other methods (doesn't need to be "main")

Reply:
The RI consistently initializes the named class (`clazz`) instead of the declaring class (`declaringClass`) for `(*env)->GetStaticMethodID(env, class, "method", ...)`. In addition to the `main` method, I verified this behaviour by passing subclasses to `GetStaticMethodID` where the method only exists in the superclass. It seems like this behaviour should be the default to match the RI.

Reply:
okay, please make that change